src/Microsoft.Diagnostics.Runtime/src/DataReaders/DbgEng/DbgEngDataReader.cs

```diff
@@ -114,36 +114,37 @@ public Architecture Architecture
         {
             get
             {
-                if (_architecture.HasValue)
-                    return _architecture.Value;
+                if (_architecture is Architecture architecture)
+                    return architecture;
 
                 IMAGE_FILE_MACHINE machineType = _control.GetEffectiveProcessorType();
                 switch (machineType)
                 {
                     case IMAGE_FILE_MACHINE.I386:
-                        _architecture = Architecture.X86;
+                        architecture = Architecture.X86;
```

Comment:
Should be _architecture.

Reply:
The local variable is assigned to the field after several lines.